airflow/providers/oracle/operators/oracle.py

```diff
@@ -93,4 +99,11 @@ def __init__(
     def execute(self, context: Context):
         self.log.info("Executing: %s", self.procedure)
         hook = OracleHook(oracle_conn_id=self.oracle_conn_id)
-        return hook.callproc(self.procedure, autocommit=True, parameters=self.parameters)
+        ti = context.get("task_instance")
+        try:
+            return hook.callproc(self.procedure, autocommit=True, parameters=self.parameters)
+        except oracledb.DatabaseError as e:
+            code_match = re.search("^ORA-(\\d+):.+", str(e))
+            if code_match and self.do_xcom_push and ti:
+                ti.xcom_push(key="ORA", value=code_match.group(1))
```

Comment:
```suggestion
            if not self.do_xcom_push or not ti:
                raise
            code_match = re.search("^ORA-(\\d+):.+", str(e))
            if code_match:
                ti.xcom_push(key="ORA", value=code_match.group(1))
```

This saves the regex match unless needed.